app/src/main/res/layout/lessons_completed_chapter_view.xml

```diff
@@ -14,51 +13,55 @@
   </data>
 
   <LinearLayout
-    android:id="@+id/chapter_container"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
+    android:orientation="vertical">
+  <LinearLayout
```

Comment:
Is it much more complicated with ConstraintLayout? It doesn't seem like it would be, but maybe I'm missing something? Generally, ConstraintLayout is preferred over nested layouts since it avoids multiple layout/measure passes (though the benefit is more noticeable when nesting relative layouts than linear).

Reply:
I've added constraint layout to the file you mentioned + one other file that has a similar xml. The other two files have a relatively complex view hierarchy and I feel they are better with the linear layout (they are actually a bit problematic to switch to the constraint layout as the nested linear layout itself plays a role). Please let me know what you think.